src/bootstrap/sanity.rs

```diff
@@ -87,6 +88,14 @@ pub fn check(build: &mut Build) {
     }
 
     if build.config.python.is_none() {
+        // set by bootstrap.py
+        match env::var("BOOTSTRAP_PYTHON") {
+            Ok(v) => build.config.python = Some(PathBuf::from(v)),
+            Err(env::VarError::NotPresent) => {}
+            Err(e) => panic!("BOOTSTRAP_PYTHON var is incorrect: {:?}", e),
```

Comment:
I don't think there's any reason for us to use `env::var` (that is, invalid UTF-8 for the python path seems fine -- we store a PathBuf). That should make this case impossible, I think.